### DISCUSSION TITLE
To update, or not to update, that is the question

### DISCUSSION THREAD

**Discussion by bracewGitHub** [tbnobody/OpenDTU#1179] [Q&A]
Frei nach William Shakespeare: 
"_To update, or not to update, that is the question_" bzw. 
"_Update oder Nichtupdate, das ist hier die Frage_"

Hallo,

ich habe seit einigen Wochen eine OpenDTU produktiv am laufen. Funktioniert bestens. Meinen Dank an alle Entwickler und Helfer!

Einen ersten Update der OpenDTU-Software habe ich bereits einige Tage nach der Inbetriebnahme erledigt. Nun wird wieder "_Neue Version verfügbar_" angezeigt.
Ich freue mich, dass die Software weiter entwickelt wird, stelle mir aber die Frage: "Sollte ich da mitziehen und auch wieder updaten oder besser die never-change-a-running-system Idee verfolgen?"

Gibt es dafür eine Regel oder Empfehlung?
Wie haltet Ihr das?

**Comment by DPO99** [comment]
Also ich denke das kann man so pauschal nicht sagen....kommt immer auf den Inhalt des Updates und die jew. genutzt Features an.

Ich persönlich nutze z.B. 2 DTU's und habe ein Balkonkraftwerk immer auf neuster Version und die 2. DTU, die auch meine Einspeisung steuert, aktualisiere ich immer nur dann, wenn wichtige Funktionen (wie MQTT z.B.) aktualisiert oder gröbere Bugs behoben werden.

Never Change a running system and never stay back ;)

**Reply by bracewGitHub** [reply]
Zitat von DPO99>"...._kommt immer auf den Inhalt des Updates und die jew. genutzt Features an_."
ja, richtig. 
Aber ich möchte auch nicht jede Woche alle Änderungen im changelog lesen, geschweige denn, dass ich als Laie die meisten nicht verstehe bzw. nicht abschätzen kann, ob zum Beispiel die Meldung "_[Merge branch 'pr1003' into dev]_" für mich und meine Gerätekonstellation zutrifft. 
Mit anderen Worten, wie kann ich abschätzen, wenn ein Update oder neues Feature für meine OpenDTU und meinen Wechselrichter relevant wird?
Wird es für die bisher verkauften Hoymiles Wechselrichter überhaupt neue "..._Features_..." geben, oder sind diese ausentwickelt?

Zitat von DPO99>"_Ich persönlich nutze z.B. 2 DTU's_ ..."
Verstehe ich Dich richtig, Du nutzt die beiden DTU's wechselweise, mit aktuellem Softwarestand am Gleichrichter und mit der vorherigen Version in Reserve, falls die aktuelle nicht zufriedenstellend funktioniert?

**Reply by DPO99** [reply]
Ja das ist eben der Unterschied zwischen ich kaufe mir eine fertige Lösung, bei der der Hersteller das Problem hat oder ich bau mir das Ding selber und bin für den Betrieb und die Wartung verantwortlich. Du wirst einen Mittelweg zwischen zu viel und zu wenig einlesen in die Materie finden müssen. Wenn es aber wirklich "krasse" Änderungen gibt, wird das eigentlich immer auf der Startseite und groß in der ReleaseNote angekündigt.

```
!! IMPORTANT UPGRADE NOTES !!
If you are upgrading from a version before 15.03.2023 you have to upgrade the partition table of the ESP32. Please follow the [this](https://github.com/tbnobody/OpenDTU/blob/master/docs/UpgradePartition.md) documentation!
```

Ich nutze insgesamt 4 HM und nutze jew. 2 auf der einen und 2 auf der anderen DTU. Die beiden unkritischen laufen immer auf neuester Version um zu sehen, ob es irgendwo rumpelt.
